src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -272,6 +272,12 @@ static void OnGetAddrInfo(DNSServiceRef sdRef, DNSServiceFlags flags, uint32_t i
 
     if (flags & kDNSServiceFlagsMoreComing)
     {
+        // If we have a timer running and we have resolved on the SRP domain but the kDNSServiceFlagsMoreComing
+        // is set, we need to cancel the timer so we don't pre-emptively call finalize().
```

Comment:
```suggestion
        // If we now don't need to have a timer while we wait for SRP results, ensure that there is no such
        // timer running.  Otherwise the timer could fire before we get the rest of the results that flags
        // say are coming, and trigger a finalize before we have all the data that is already available.
```